homeassistant/components/ista_ecotrend/coordinator.py

```diff
@@ -0,0 +1,76 @@
+"""DataUpdateCoordinator for Ista EcoTrend integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyecotrend_ista.exception_classes import (
+    InternalServerError,
+    KeycloakError,
+    LoginError,
+    ServerError,
+)
+from pyecotrend_ista.pyecotrend_ista import PyEcotrendIsta
+from requests.exceptions import RequestException
+
+from homeassistant.const import CONF_EMAIL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IstaCoordinator(DataUpdateCoordinator):
+    """Ista EcoTrend data update coordinator."""
+
+    details: dict[str, Any]
+
+    def __init__(self, hass: HomeAssistant, ista: PyEcotrendIsta) -> None:
+        """Initialize ista EcoTrend data update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(days=1),
+        )
+        self.ista = ista
+        self.details = {}
+
+    async def _async_update_data(self):
+        data: dict[str, Any] = {}
+        try:
+            if not self.details:
+                result = await self.hass.async_add_executor_job(
+                    self.ista.get_consumption_unit_details
+                )
+                self.details = {
+                    consumption_unit: next(
+                        details
+                        for details in result["consumptionUnits"]
+                        if details["id"] == consumption_unit
+                    )
+                    for consumption_unit in self.ista.getUUIDs()
```

Comment:
can this throw? otherwise move out of `try`

Reply:
maybe IndexError or KeyError if something is wrong in the json response but didn't happen till now